rapidfuzz/details/types.hpp

```diff
@@ -126,16 +126,16 @@ inline bool operator==(EditOp a, EditOp b)
  */
 struct Opcode {
     EditType type;             /**< type of the edit operation */
-    std::ptrdiff_t src_begin;  /**< index into the source string */
-    std::ptrdiff_t src_end;    /**< index into the source string */
-    std::ptrdiff_t dest_begin; /**< index into the destination string */
-    std::ptrdiff_t dest_end;   /**< index into the destination string */
+    std::size_t src_begin;  /**< index into the source string */
+    std::size_t src_end;    /**< index into the source string */
+    std::size_t dest_begin; /**< index into the destination string */
+    std::size_t dest_end;   /**< index into the destination string */
```

Comment:
As a note to myself: This is strictly speaking backwards compatible (already the change to std::ptrdiff_t), so this will require a major version bump.

Reply:
I tested these changes with MSVC 2022. They did not reintroduce warnings.

Reply:
Hm I would assume something like:
```
a.begin() + op.src_begin
```
would now cause warning. Otherwise casts like `j2len_.begin() + static_cast<std::ptrdiff_t>(b_low)` wouldn't be required either. Granted this would at worst introduce warnings, but definitely not break the user code. So this might or might not be counted as breaking change.